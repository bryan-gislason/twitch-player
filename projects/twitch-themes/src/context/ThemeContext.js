import PropTypes from "prop-types";
import React, { createContext, useState } from "react";
import { light } from "../themes";

const defaultTheme = light;

export const ThemeContext = createContext();

export const ThemeProvider = ({ theme, ...props }) => {
  const themeState = useState(theme || defaultTheme);

  return <ThemeContext.Provider {...props} value={themeState} />;
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};
