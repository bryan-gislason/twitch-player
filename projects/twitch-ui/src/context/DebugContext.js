import PropTypes from "prop-types";
import React, { createContext, useState } from "react";

const defaultValue = false;

export const DebugContext = createContext();

export const DebugProvider = ({ value, ...props }) => {
  const debugState = useState(value || defaultValue);

  return <DebugContext.Provider {...props} value={debugState} />;
};

DebugProvider.propTypes = {
  value: PropTypes.bool,
};
