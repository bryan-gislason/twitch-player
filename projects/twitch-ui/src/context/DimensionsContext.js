import React, { createContext, useState } from "react";

const defaultDimensions = {};

export const DimensionsContext = createContext();

export const DimensionsProvider = ({ ...props }) => {
  const dimensionsState = useState(defaultDimensions);

  return <DimensionsContext.Provider {...props} value={dimensionsState} />;
};
