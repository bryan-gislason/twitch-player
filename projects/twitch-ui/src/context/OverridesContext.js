import PropTypes from "prop-types";
import React, { createContext } from "react";

const defaultOverrides = {
  StreamPlayer: () => null,
};

export const OverridesContext = createContext(defaultOverrides);

export const OverridesProvider = ({ overrides, ...props }) => (
  <OverridesContext.Provider {...props} value={overrides || defaultOverrides} />
);

OverridesProvider.propTypes = {
  overrides: PropTypes.object,
};
