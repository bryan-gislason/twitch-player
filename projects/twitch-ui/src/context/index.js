export * from "./DebugContext";
export * from "./DimensionsContext";
export * from "./OverridesContext";
