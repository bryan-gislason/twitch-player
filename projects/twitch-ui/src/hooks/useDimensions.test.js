import React from "react";
import { Text, View } from "react-native";
import { fireEvent, render } from "../testing";
import { DimensionsProvider } from "../context";
import { useDimensions } from ".";

const SomeComponent = () => {
  const [dimensions, updateDimensions] = useDimensions("foobar");

  return (
    <View onLayout={updateDimensions}>
      <Text>{dimensions && `${dimensions.width}x${dimensions.height}`}</Text>
    </View>
  );
};

describe("useDimensions", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <DimensionsProvider>
        <SomeComponent />
      </DimensionsProvider>
    );
  });

  it("renders without dimensions", () => {
    expect(instance.asFragment()).toMatchSnapshot();
  });

  describe("when the dimensions change", () => {
    beforeEach(() => {
      fireEvent.resize(window, { target: { width: 480, height: 640 } });
    });

    it("renders with the updated dimensions", () => {
      expect(instance.asFragment()).toMatchSnapshot();
    });
  });
});
