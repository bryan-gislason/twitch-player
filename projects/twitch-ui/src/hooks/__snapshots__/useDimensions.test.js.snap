// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDimensions renders without dimensions 1`] = `
<DocumentFragment>
  <div
    class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    />
  </div>
</DocumentFragment>
`;

exports[`useDimensions when the dimensions change renders with the updated dimensions 1`] = `
<DocumentFragment>
  <div
    class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      480x640
    </div>
  </div>
</DocumentFragment>
`;
